### PR TITLE
treewide: Remove H20 EngB and L15 EngA

### PR DESCRIPTION
**Pull request by carlescufi**
Bring in changes in:
https://github.com/nrfconnect/sdk-zephyr/pull/2147

### CODE REVIEW

**Review by nordicjm — APPROVED** [review]

**Review by tomi-font — APPROVED** [review]
`subsys/nrf_security`

**Review by rugeGerritsen — APPROVED** [review]

File: samples/bluetooth/radio_notification_cb/sample.yaml [hunk 1]
```diff
@@ -9,12 +9,10 @@ tests:
       - nrf52dk/nrf52832
       - nrf52840dk/nrf52840
       - nrf5340dk/nrf5340/cpunet
-      - nrf54l15pdk/nrf54l15/cpuapp
```
Comment (nordic-segl) [inline samples/bluetooth/radio_notification_cb/sample.yaml:12]: There is no nrf54l15dk/nrf54l15/cpuapp on the list. It this expected?

File: samples/mpsl/timeslot/sample.yaml [hunk 2]
```diff
@@ -9,8 +9,7 @@ tests:
       - nrf52dk/nrf52832
       - nrf52840dk/nrf52840
       - nrf5340dk/nrf5340/cpunet
-      - nrf54l15pdk/nrf54l15/cpuapp
```
Comment (nordic-segl) [inline samples/mpsl/timeslot/sample.yaml:12]: There is no nrf54l15dk/nrf54l15/cpuapp on the list. It this expected?

File: samples/bluetooth/radio_notification_cb/sample.yaml (hunk 1 above)
Comment (carlescufi) [reply samples/bluetooth/radio_notification_cb/sample.yaml:12]: @rugeGerritsen ^^

File: samples/mpsl/timeslot/sample.yaml (hunk 2 above)
Comment (carlescufi) [reply samples/mpsl/timeslot/sample.yaml:12]: @rugeGerritsen ^^

File: samples/bluetooth/radio_notification_cb/sample.yaml (hunk 1 above)
Comment (rugeGerritsen) [reply samples/bluetooth/radio_notification_cb/sample.yaml:12]: It should be added

File: samples/mpsl/timeslot/sample.yaml (hunk 2 above)
Comment (rugeGerritsen) [reply samples/mpsl/timeslot/sample.yaml:12]: It should be added here as well

**Review by annwoj — APPROVED** [review]

**Review by nordic-piks — APPROVED** [review]

**Review by nordic-segl — APPROVED** [review]

**Comment by nordic-segl** [comment]
I see following error in the Twister stage:
```
18:11:39    File "/home/ncs/toolchains/b77d8c1312/usr/local/lib/python3.12/pathlib.py", line 1240, in resolve
18:11:39      s = self._flavour.realpath(self, strict=strict)
18:11:39          ^^^^^^^^^^^^^
18:11:39  AttributeError: 'str' object has no attribute '_flavour'
```

File: subsys/nrf_security/src/drivers/cracen/cracenpsa/src/lib_kmu.c [hunk 3]
```diff
@@ -68,12 +68,8 @@ int lib_kmu_provision_slot(int slot_id, struct kmu_src *kmu_src)
...
 	}
 
-	/* The DEST address must be 64-bit aligned on nrf54l15pdk.
-	 * On nrf54l15dk, the alignment requirement changed to 128 bits.
-	 */
-	__ASSERT(IS_PTR_ALIGNED_BYTES(kmu_src->dest,
-				      (IS_ENABLED(CONFIG_SOC_NRF54L15_ENGA_CPUAPP) ? 8 : 16)),
-		 "DEST misaligned");
+	 /* On nrf54l15dk, the alignment requirement is 128 bits. */
+	__ASSERT(IS_PTR_ALIGNED_BYTES(kmu_src->dest 16), "DEST misaligned");
```
Comment (nordic-segl) [inline subsys/nrf_security/src/drivers/cracen/cracenpsa/src/lib_kmu.c:72]: missing comma before 16??
__ASSERT(IS_PTR_ALIGNED_BYTES(kmu_src->dest, 16), "DEST misaligned");

**Review by rugeGerritsen — APPROVED** [review]

**Review by shanthanordic — APPROVED** [review]

**Review by maje-emb — APPROVED** [review]

**Review by koffes — APPROVED** [review]

**Review by ankuns — APPROVED** [review]

**Review by miha-nordic — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]
Manifest can be updated

**Review by rlubos — APPROVED** [review]

**Pull request merged**